### ISSUE TITLE
shinyApp() generates error resource_folder not found.

### ISSUE THREAD

**Issue by paullemmens** [rstudio/rmarkdown#619]
I have a fairly simple shinyApp that I am embedding in an rmarkdown file as described [here](http://rmarkdown.rstudio.com/authoring_embedded_shiny.html). The app is as follows: 

```
```{r timeseries, echo = FALSE}
shinyApp(
  ui = fluidPage(
    sidebarLayout(
      sidebarPanel(selectInput('pp.id', label = 'Select participant:', choices = unique(aee$pp))), 
      mainPanel(plotOutput("tseries", dblclick = 'plot_dblclick', 
                           brush = brushOpts(id = 'plot_brush', resetOnNew = TRUE)))
    )
  ), 

  server = function(input, output) {
    ranges <- reactiveValues(x = NULL, y = NULL) 
    output$tseries <- renderPlot({
        ggplot(data = subset(aee, pp == input$pp.id),
               mapping = aes(x = t.stamp, y = aee, colour = algo)) +
          geom_line() +
          coord_cartesian(xlim = ranges$x, ylim = ranges$y) + mytheme
    })

    observeEvent(input$plot_dblclick, {
      brush <- input$plot_brush
      if (!is.null(brush)) {
        ranges$x <- c(brush$xmin, brush$xmax)
        ranges$y <- c(brush$ymin, brush$ymay)
      } else {
        ranges$x <- NULL
        ranges$y <- NULL
      }
    })
  },

  options = list(height = 300)
)
```

When I run the markdown file with `run('my.Rmd')` it stops with the error `Error: object 'resource_folder' not found`. The R console generates the following trace after stopping the run() call with Esc. No clue what is going on and haven't been able to track down something relevant on the web. I would appreciate a bit of help to get this solved.

```
label: timeseries (with options) 
List of 1
 $ echo: logi FALSE


output file: C:/Users/NLV19419/AppData/Local/Temp/RtmpKOys0A/3.timeseries.knit.md

"C:/Users/NLV19419/AppData/Local/Pandoc/pandoc" +RTS -K512m -RTS "C:/Users/NLV19419/AppData/Local/Temp/RtmpKOys0A/3.timeseries.utf8.md" --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash-implicit_figures --output "C:/Users/NLV19419/AppData/Local/Temp/RtmpKOys0A/file32a450086bf5.html" --smart --email-obfuscation none --standalone --section-divs --template "C:\PROGRA~1\R\R-32~1.3\library\RMARKD~1\rmd\h\DEFAUL~1.HTM" --variable "theme:bootstrap" --mathjax --variable "mathjax-url:https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" --no-highlight --variable highlightjs=file32a450086bf5_files/highlight 

Output created: C:/Users/NLV19419/AppData/Local/Temp/RtmpKOys0A/file32a450086bf5.html
Warning: Error in dir.exists: object 'resource_folder' not found
Stack trace (innermost first):
    87: dir.exists
    86: utils::file_test
    85: dir_exists
    84: reactive reactive({
    out <- rmd_cached_output(file, encoding)
    output_dest <- out$dest
    if (out$cached) {
        if (nchar(out$resource_folder) > 0) {
            shiny::addResourcePath(basename(out$resource_folder), 
                out$resource_folder)
        }
        return(out$shiny_html)
    }
    if (!file.exists(dirname(output_dest))) {
        dir.create(dirname(output_dest), recursive = TRUE, mode = "0700")
    }
    resource_folder <- knitr_files_dir(output_dest)
    perf_timer_reset_all()
    dependencies <- list()
    shiny_dependency_resolver <- function(deps) {
        dependencies <<- deps
        list()
    }
    output_opts <- list(self_contained = FALSE, copy_resources = TRUE, 
        dependency_resolver = shiny_dependency_resolver)
    message("\f")
    args <- merge_lists(list(input = reactive_file(), output_file = output_dest, 
        output_dir = dirname(output_dest), output_options = output_opts, 
        intermediates_dir = dirname(output_dest), runtime = "shiny"), 
        render_args)
    result_path <- shiny::maskReactiveContext(do.call(render, 
        args))
    if (!dir_exists(resource_folder)) 
        dir.create(resource_folder, recursive = TRUE)
    shiny::addResourcePath(basename(resource_folder), resource_folder)
    dependencies <- append(dependencies, list(create_performance_dependency(resource_folder)))
    write_deps <- base::file(file.path(resource_folder, "shiny.dep"), 
        open = "wb")
    on.exit(close(write_deps), add = TRUE)
    serialize(dependencies, write_deps, ascii = FALSE)
    if (!isTRUE(out$cacheable)) {
        shiny::onReactiveDomainEnded(shiny::getDefaultReactiveDomain(), 
            function() {
                unlink(result_path)
                unlink(resource_folder, recursive = TRUE)
            })
    }
    shinyHTML_with_deps(result_path, dependencies)
})
    73: doc
    72: shiny::renderUI
    71: func
    70: output$__reactivedoc__
     3: <Anonymous>
     2: do.call
     1: rmarkdown::run
```

**Comment by jjallaire** [comment]
@jmcphers Any idea on this one?

**Comment by jmcphers** [comment]
@paullemmens, can you paste the whole document that's having a problem (or at least the YAML header)? What version of the rmarkdown package are you using?

I wasn't able to reproduce this locally with your code; I got a Shiny app up and running without issue (although I did have to replace the data variables with nulls).

Does the Shiny document template work for you? (New File -> R Markdown -> Shiny -> Shiny Document)

**Comment by paullemmens** [comment]
@jmcphers See the whole document below, including the `sessionInfo()`. The main thing that I did not mention was indeed that I am running this in the good ol' console instead of in rstudio. Note that I am changing the file from a `knitrBootstrap` styled document to this one using (just) `rmarkdown`. I presume that the additional chunk stuff referring to bootstrap does not interfere.

```
R version 3.2.3 (2015-12-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reshape2_1.4.1  dplyr_0.4.3     scales_0.3.0    stringr_1.0.0   Cairo_1.5-9     shiny_0.13.0    rmarkdown_0.9.2 ggplot2_2.0.0  

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.2      knitr_1.11.9     magrittr_1.5     munsell_0.4.2    colorspace_1.2-6 xtable_1.8-0     R6_2.1.1         plyr_1.8.3       tools_3.2.3      parallel_3.2.3   grid_3.2.3      
[12] gtable_0.1.2     DBI_0.3.1        htmltools_0.3    assertthat_0.1   yaml_2.1.13      digest_0.6.8     formatR_1.2.1    evaluate_0.8     mime_0.4         stringi_1.0-1    jsonlite_0.9.19 
[23] httpuv_1.3.3    
```

    ---
    title: 'Time series plots with zoom via Shiny'
    output: html_document
    runtime: shiny
    ---

    ```{r set-options-read-chunks, echo=FALSE, message=FALSE, warning=FALSE,         bootstrap.show.code = FALSE}
    knitr::opts_chunk$set(results = 'markup', echo=TRUE, message=FALSE, warning=FALSE, dev=c('CairoSVG'), 
                          bootstrap.show.code=FALSE)
    rm(list=ls())
    options(width=120)
    require(Cairo)
    require(stringr)
    require(ggplot2)
    require(scales)
    require(dplyr)
    require(reshape2)
    require(shiny)
    mygeoms(use=FALSE)
    palette <- 'Dark2'
    #palette <- 'Paired'
    ```

    Plain text to explain stuff.

    ```{r load-data}
    data.dir <- '//code1.emi.philips.com/storage/2013-0176_PREVIEW/PrEvIeW 2013-0176/WP on   Measurement/Physical Activity study/Data/Data'
    aee.file <- file.path(data.dir, 'aee_timeseries_ms.RData')
    load(aee.file)
    ```

    ```{r timeseries, echo = FALSE}
    shinyApp(
      ui = fluidPage(
        sidebarLayout(
          sidebarPanel(selectInput('pp.id', label = 'Select participant:', choices = unique(aee$pp))), 
          mainPanel(plotOutput("tseries", dblclick = 'plot_dblclick', 
                               brush = brushOpts(id = 'plot_brush', resetOnNew = TRUE)))
        )
      ), 

      server = function(input, output) {
        ranges <- reactiveValues(x = NULL, y = NULL) 
        output$tseries <- renderPlot({
            ggplot(data = subset(aee, pp == input$pp.id),
                   mapping = aes(x = t.stamp, y = aee, colour = algo)) +
              geom_line() +
              coord_cartesian(xlim = ranges$x, ylim = ranges$y) + mytheme
        })

        observeEvent(input$plot_dblclick, {
           brush <- input$plot_brush
          if (!is.null(brush)) {
            ranges$x <- c(brush$xmin, brush$xmax)
            ranges$y <- c(brush$ymin, brush$ymay)
           } else {
            ranges$x <- NULL
            ranges$y <- NULL
          }
        })
      },
    
      options = list(height = 300)
    )
    ```

**Comment by jmcphers** [comment]
Thanks for posting the whole document; with that I was able to reproduce the problem. It looks like it's caused by this line in the setup chunk:

    rm(list=ls())

The environment in which that chunk executes includes some state required by the interactive doc; the `resource_folder` is among the values being removed. 

This is arguably a bug in rmarkdown (which should ideally not expose these internal values to the document being rendered), but to work around it until it's fixed you can probably just axe that line.
